### ISSUE TITLE
[METEOR@1.2-rc.15] Scss compiler error: file to import not found or unreadable

### ISSUE THREAD

**Issue by gerwinbrunner** [meteoric/ionic-sass#16]
It seems that the package assets are handled differently in the upcoming meteor 1.2 release.

```
   While building the application:

   .../app/client/stylesheets/main.sass:20:9:
   Scss compiler error: file to import not found or unreadable:
   .meteor/local/build/programs/server/assets/packages/meteoric_ionic-sass/ionic
   Current dir:
   .../app/client/stylesheets/

```

I get this error no matter how often I restart. 
It seems that meteor does not copy the package anymore to the assets folder here:
`.meteor/local/build/programs/server/assets/packages/meteoric_ionic-sass`

The meteor upgrade notes mention a new way to specify assets for packages. See here:
https://quip.com/RXFlAk9Rc2xI

**Comment by gerwinbrunner** [comment]
Any idea?

**Comment by mountainash** [comment]
Also getting the same error now that 1.2.0.1 is released.

As a work around I've copied the 2 needed files from the old build dir to my /public dir (but obviously this is not ideal as it will outdate as it misses updates).

**Comment by gwendall** [comment]
This is the reason: https://github.com/meteor/meteor/wiki/Breaking-changes-in-Meteor-1.2#build-tool-and-package-improvements. 

Change [this line](https://github.com/meteoric/ionic-sass/blob/master/package.js#L50) to `api.addAssets('_ionic.scss')`. Same idea for [meteoric:ionicons-sass](https://github.com/meteoric/ionicons-sass/blob/master/package.js#L27). 

We will update the packages accordingly.

**Comment by kevohagan** [comment]
Any news on this? I tried but havent succeeded..

**Comment by austingayler** [comment]
@gwendall is there an update on this?

**Comment by kevohagan** [comment]
what ive done in the mean time @austingayler is to put all my ionic and
ionicons scss files in my public folder and imported them from there.

**Comment by myktra** [comment]
+1. Would be great to have this updated for Meteor's new build system in 1.2 as soon as reasonably possible!

**Comment by elie222** [comment]
@gwendall merely changing that line doesn't fix the problems

**Comment by austingayler** [comment]
I have not been having this problem, but a few coworkers have (maybe windows/osx problem?--I have windows). They both solved it different ways:
try

```
meteor remove fourseven:scss
meteor remove meteoric:ionic-sass
meteor remove meteoric:ionicons-sass
meteor add fourseven:scss@2.0.0
meteor add meteoric:ionic-sass
meteor add meteoric:ionicons-sass 
```

and

```
Run Meteor
Comment out the
@import '.meteor/local/build/programs/server/assets/packages/meteoric_ionic-sass/_ionic';
@import '.meteor/local/build/programs/server/assets/packages/meteoric_ionicons-sass/_ionicons';
Let it rebuild -you should get an app with no styling
Then comment back in and the project will load correctly 
```

**Comment by elie222** [comment]
Thanks @austingayler. Working for me now. That was taking far too long to fix.

**Comment by austingayler** [comment]
@elie222 sweet, glad to help.

**Comment by sebakerckhof** [comment]
This is due to changes in meteor 1.2 and the latest version of fourseven:scss (note: there are errors in 3.3.3, better use 3.4.0-beta1) , besides caching, this supports the cross-package imports similar to the updated less and stylus packages.

The files need to be added to the client using addFiles with isImport flag set to true (although not necessary for files that are prefixed with an underscore)

**Comment by knowself** [comment]
Thanks @austingayler. That really sucked. I don't even like having windows being my primary dev box, but I didn't have the issue and my partner was having trouble with his Mac. That was frustrating!

**Comment by hashwin** [comment]
This is still broken when deploying to heroku, scalingo etc. Works on the regular meteor deploy though.

**Comment by sebakerckhof** [comment]
@hashwin, see if this package works (please read the instructions on the package page): 
https://atmospherejs.com/seba/ionic-sass

**Comment by hashwin** [comment]
@sebakerckhof Yeah, that worked to compile, but the navigation animations and ionicons seem to be broken. Looks like it works if you make the same changes you made, but using this repo as a source rather than https://github.com/driftyco/ionic

**Comment by sebakerckhof** [comment]
Ow, I think I forgot to add the ionicons font files. Don't know why the animations don't work though. This could be a change between ionic 1.0-beta13 used here and 1.1.0 that I use.

**Comment by sebakerckhof** [comment]
I've published a new version that fixes icons.

**Comment by daveeel** [comment]
@sebakerckhof 
Just trying to use https://atmospherejs.com/seba/ionic-sass ( seba:ionic-sass@1.1.0
 ) but icons are missing. Is 1.1.0 the latest version?

I'm using meteoric:ionic & trying to replace meteoric:ionic-sass & meteoric:ionicons-sass which bugs me upon compiling my app

**Comment by hashwin** [comment]
@daveeel I have created a fork of this repo with @sebakerckhof's changes here: https://atmospherejs.com/hashwin/ionic-sass and https://atmospherejs.com/hashwin/ionicons-sass

**Comment by daveeel** [comment]
@hashwin Used your fork, works great :+1:

**Comment by krishnaff** [comment]
For some weird reason, I was getting a { } between the file path (Meteor 1.2) :

```
While processing files with fourseven:scss (for target web.browser):
   /client/styles.scss: Scss compiler error: File to read not found or unreadable:
   /Users/insaneinc/projects/projectname/{}/client/styles.scss
```

This is the only thing that worked for me 
`meteor add fourseven:scss@=3.2.0`

**Comment by charlieman** [comment]
Thanks @insaneinc01 downgrading to 3.2.0 (from 3.3.3_3) worked for me as well.

**Comment by gbhatnag** [comment]
Simply downgrading to `fourseven:scss@=3.2.0` worked wonders for me.  I'm using `Meteor 1.2.0.2`.  Thanks @insaneinc01 and all on this thread for debugging.  I've been banging my head against this for a long while.

**Comment by evanglerm** [comment]
This "Simply downgrading to fourseven:scss@=3.2.0 worked wonders for me. I'm using Meteor 1.2.0.2. Thanks @insaneinc01 and all on this thread for debugging. I've been banging my head against this for a long while."

AND

Commenting Import stmt then putting it back worked for me. 

Thanks Guys.

**Comment by avishaan** [comment]
Oh wow, thanks for this thread. Downgrading worked for me. I wished I had seen this a couple days earlier though. I ended up just building a native app instead but at least for my next one maybe i'll built it with this again.

**Comment by rdewolff** [comment]
Thanks for @insaneinc01 tip, downgrading with `meteor add fourseven:scss@=3.2.0` worked like a charm!

Cheers!

**Comment by mervynteo** [comment]
I upgraded to Meteor 1.2+ and tried the following and nothing worked 

> - downgrade to fourseven 3.2 
> - tried to move the asset folders but realised nothing in there?

The issue is when it builds, Meteor no longers inserts those scss files for ionic and ionicon. When i navigate to the folders, those files arent there too.

While i have the public cdn link for ionicons which worked, I dont have the same public link for Ionic.sass.

**Comment by krishnaff** [comment]
A `meteor reset` and `rm -rf ~/.cordova` sometimes helps.

**Comment by hashwin** [comment]
@mervynteo see if using these repos works?
https://atmospherejs.com/hashwin/ionic-sass and https://atmospherejs.com/hashwin/ionicons-sass

**Comment by hosan** [comment]
@hashwin 's solution works :+1:

**Comment by mervynteo** [comment]
cool ...hashwin's packages works.. thanks alot @hashwin!

**Comment by mervynteo** [comment]
Thanks! @hashwin the packages work perfect...a real life saver! :+1:

**Comment by thatgibbyguy** [comment]
Will confirm, using `fourseven:scss@=3.2.0` fixed this issue for me.

**Comment by iuraya** [comment]
Thank you @insaneinc01. The downgrade works perfectly!

**Comment by sferoze** [comment]
@hashwin thanks your packages work as specified!

**Comment by josephmiclaus** [comment]
thanks @hashwin !
